OSDOCS 5806: Adding docs for using the secrets store CSI driver (#62385)
openshift/openshift-docs: modules/persistent-storage-csi-secrets-store-driver-overview.adoc

// Module included in the following assemblies:
//
// * storage/container_storage_interface/persistent-storage-csi-secrets-store.adoc
// * nodes/pods/nodes-pods-secrets-store.adoc

> ifeval::["{context}" == "persistent-storage-csi-secrets-store"]

Reviewer: Can we just call it Secret Store CSI Driver?

Reviewer 2: I'm not sure whether this name is user-visible, but it is a bit misleading because secrets store is one of the rare CSI drivers that does not provide any persistent storage (only epehemeral volumes).

Author: Yeah this is just file/ID name, nothing really visible to users. And the only reason this one is named differently is because I nabbed [the file from Lisa's PR](https://github.com/openshift/openshift-docs/pull/60607/files#diff-6478407566af3921b0b2f6eba0d242a66300b5f54bbc395a5b80b9accd8b31bf) to work on this "about" content more. I'm not sure whether this file will go in by her PR or mine yet, that is still TBD.

But I believe she chose this naming scheme because of how the other CSI driver docs are named. CC @lpettyjo to consider adjusting the naming scheme for the files in your PR based on this feedback.

// Module included in the following assemblies:
//
// * storage/container_storage_interface/persistent-storage-csi-secrets-store.adoc
// * nodes/pods/nodes-pods-secrets-store.adoc

ifeval::["{context}" == "persistent-storage-csi-secrets-store"]
:storage:
endif::[]
ifeval::["{context}" == "nodes-pods-secrets-store"]
:nodes:
endif::[]

:_content-type: CONCEPT
[id="persistent-storage-csi-secrets-store-driver-overview_{context}"]
ifdef::storage[]
= Overview
endif::storage[]
ifdef::nodes[]
= About the {secrets-store-operator}
endif::nodes[]

Kubernetes secrets are stored with Base64 encoding. etcd provides encryption at rest for these secrets, but when secrets are retrieved, they are decrypted and presented to the user. If role-based access control is not configured properly on your cluster, anyone with API or etcd access can retrieve or modify a secret. Additionally, anyone who is authorized to create a pod in a namespace can use that access to read any secret in that namespace.

> For secure storage and management of your secrets, the {product-title} Secrets Store Container Storage Interface (CSI) Driver Operator allows you to mount secrets from an external secret management system, such as Azure Key Vault, using a provider plugin. Applications can use the secret, but the secret does not persist on the system after the application pod is destroyed.

Reviewer 3: ```suggestion
To store and manage your secrets securely, you can configure the {product-title} Secrets Store Container Storage Interface (CSI) Driver Operator to mount secrets from an external secret management system, such as Azure Key Vault, by using a provider plugin. Applications can then use the secret, but the secret does not persist on the system after the application pod is destroyed.
```
- Edit to remove `allows you to` (anthropomorphism)

Author: @lpettyjo Checking since this was your content - are you okay with me making these recommended changes?

Reviewer 4: @bergerhoffer fine by me.

> The {secrets-store-operator}, `secrets-store.csi.k8s.io`, allows {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as a volume. The {secrets-store-operator} communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container's file system. Secrets store volumes are mounted in-line.

Reviewer 3: ```suggestion
The {secrets-store-operator}, `secrets-store.csi.k8s.io`, enables {product-title} to mount multiple secrets, keys, and certificates stored in enterprise-grade external secrets stores into pods as a volume. The {secrets-store-operator} communicates with the provider using gRPC to fetch the mount contents from the specified external secrets store. After the volume is attached, the data in it is mounted into the container's file system. Secrets store volumes are mounted in-line.
```
- Maybe `enables` is better here because the subject is adding functionality and not giving permission.

Author: @lpettyjo Okay with this change?

Reviewer 4: @bergerhoffer fine by me.
